File: Batu Gunting Kertas/BatuGuntingKertas.java
import java.util.Scanner;
import java.util.Random;

public class BatuGuntingKertas
{
    static String [] choices = {"Batu", "Gunting", "Kertas"};
    static int [][] results = {{0, 1, -1}, {-1, 0, 1}, {1, -1, 0}};
    static Scanner scan = new Scanner (System.in);
    static Random rand = new Random ();

    public static void main (String [] args)
    {
        while (true)
        {
            int PlayerChoice, ComputerChoice;
        
            PlayerChoice = player ();
            ComputerChoice = computer ();
            challenge (PlayerChoice, ComputerChoice);

            System.out.print ("\nIngin bermain lagi? (Y/y atau ketik sembarang untuk keluar): ");
            char choice = scan.next ().charAt (0);

            if (Character.toUpperCase (choice) != 'Y')
            {
                System.out.println ("Terima kasih telah bermain!");
                break;
            }
        }
    }

    static int player ()
    {
        System.out.println ("====================");
        System.out.println ("Batu-Gunting-Kertas");
        System.out.println ("====================");
        System.out.println ("1. Batu");
        System.out.println ("2. Gunting");
        System.out.println ("3. Kertas");
        System.out.println ("====================");
        System.out.print ("Pilih salah satu: ");
        int PlayerChoice = scan.nextInt ();

        while (PlayerChoice < 1 || PlayerChoice > 3)
        {
            System.out.println ("Masukkan hanya angka 1-3");
            System.out.print ("Pilih salah satu: ");
            PlayerChoice = scan.nextInt ();
        }

        System.out.printf ("\nPlayer memilih [%s]\n",choices [PlayerChoice - 1]);
        return PlayerChoice;
    }

    static int computer ()
    {
        int ComputerChoice = rand.nextInt (3);
        System.out.printf ("Komputer memilih [%s]\n", choices [ComputerChoice]);
        return ComputerChoice;
    }

    static void challenge (int PlayerChoice, int ComputerChoice)
    {
        int result = results [PlayerChoice - 1][ComputerChoice];
        if (result == 0) System.out.println ("Seri");
        else if (result == 1) System.out.println ("Player Menang, Computer Kalah");
        else System.out.println ("Player Kalah, Computer Menang");
    }
}

File: Batu Gunting Kertas/Batu-Gunting-Kertas.cpp
#include <iostream>

char pilihan_pemain ();
char pilihan_komputer ();
void tampilkan_pilihan (char pilihan);
void pemenang (char pemain, char komputer);

int main ()
{
    char pemain, komputer;

    pemain = pilihan_pemain ();

    std::cout << "Pilihanmu: ";
    tampilkan_pilihan (pemain);

    komputer = pilihan_komputer ();
    std::cout << "Pilihan komputer: ";
    tampilkan_pilihan (komputer);

    pemenang (pemain, komputer);

    return 0;
}

char pilihan_pemain ()
{
    char pemain;

    do
    {
        std::cout << "===================\n";
        std::cout << "Batu_Gunting-Kertas\n";
        std::cout << "===================\n";
        std::cout << "'b' untuk batu\n";
        std::cout << "'g' untuk gunting\n";
        std::cout << "'k' untuk kertas\n";
        std::cout << "Silahkan pilih: ";
        std::cin >> pemain;
    } while (pemain != 'b' && pemain != 'g' && pemain != 'k');

    return pemain;
}

char pilihan_komputer ()
{
    srand (time (NULL));
    unsigned short angka = rand () % 3 + 1;

    switch (angka)
    {
        case 1: return 'b';
                break;
        case 2: return 'g';
                break;
        case 3: return 'k';
                break;
    }
}

void tampilkan_pilihan (char pilihan)
{
    switch (pilihan)
    {
        case 'b':
            std::cout << "Batu\n";
            break;
        case 'g':
            std::cout << "Gunting\n";
            break;
        case 'k':
            std::cout << "Kertas\n";
            break;
    }
}

void pemenang (char pemain, char komputer)
{
    switch (pemain)
    {
        case 'b':
            if (komputer == 'b')
            {
                std::cout << "Seri!";
            }

            else if (komputer == 'g')
            {
                std::cout << "Kamu menang!";
            }

            else if (komputer == 'k')
            {
                std::cout << "Kamu kalah!";
            }
            break;

        case 'g':
            if (komputer == 'b')
            {
                std::cout << "Kamu kalah!";
            }

            else if (komputer == 'g')
            {
                std::cout << "Seri!";
            }

            else if (komputer == 'k')
            {
                std::cout << "Kamu menang!";
            }
            break;

        case 'k':
            if (komputer == 'b')
            {
                std::cout << "Kamu menang!";
            }

            else if (komputer == 'g')
            {
                std::cout << "Kamu kalah!";
            }

            else if (komputer == 'k')
            {
                std::cout << "Seri!";
            }
            break;
    }
}